### PR TITLE
Add support for field level number to str coercion option

### LINKED ISSUE #8383: coerce_numbers_to_str needs a per-field variant
### Initial Checks

- [X] I have searched Google & GitHub for similar requests and couldn't find anything
- [X] I have read and followed [the docs](https://docs.pydantic.dev) and still think this feature is missing

### Description

We are migrating our code from Pydantic 1 to 2, and came across a situation where we do want some fields to coerce numbers to strings, but other fields in the model should not be coerced. As far as I can tell from the docs, `coerce_numbers_to_str` is an all-or-nothing thing, and will affect the entire model. Being able to specify fields would be nice. Maybe something like:
```
model_config = ConfigDict(coerce_numbers_to_str="field1,field2,field3")
```
I don't know if that's the best syntax, but that would serve our purposes.

If there is a way to work around this feature not being there, could it be added to https://docs.pydantic.dev/latest/api/config/#pydantic.config.ConfigDict.coerce_numbers_to_str ?

### Affected Components

- [ ] [Compatibility between releases](https://docs.pydantic.dev/changelog/)
- [X] [Data validation/parsing](https://docs.pydantic.dev/concepts/models/#basic-model-usage)
- [ ] [Data serialization](https://docs.pydantic.dev/concepts/serialization/) - `.model_dump()` and `.model_dump_json()`
- [ ] [JSON Schema](https://docs.pydantic.dev/concepts/json_schema/)
- [ ] [Dataclasses](https://docs.pydantic.dev/concepts/dataclasses/)
- [X] [Model Config](https://docs.pydantic.dev/concepts/config/)
- [ ] [Field Types](https://docs.pydantic.dev/api/types/) - adding or changing a particular data type
- [ ] [Function validation decorator](https://docs.pydantic.dev/concepts/validation_decorator/)
- [ ] [Generic Models](https://docs.pydantic.dev/concepts/models/#generic-models)
- [ ] [Other Model behaviour](https://docs.pydantic.dev/concepts/models/) - `model_construct()`, pickling, private attributes, ORM mode
- [ ] [Plugins](https://docs.pydantic.dev/) and integration with other tools - mypy, FastAPI, python-devtools, Hypothesis, VS Code, PyCharm, etc.

### PR DESCRIPTION
**Pull request by NeevCohen**
## Change Summary

This change adds support for field level number to str coercion option as requested in #8383.

Support in pydantic-core has already been implemented in pydantic/pydantic-core@c607fd82bed154a473b13c2e42a9b6b71b1c945d.

This change requires a new pydantic-core version which includes the above commit. Opening this PR now to remember to merge it once a new core version comes out.

## Related issue number

Fix #8383

## Checklist

* [x] The pull request title is a good summary of the changes - it will be used in the changelog
* [x] Unit tests for the changes exist
* [ ] Tests pass on CI
* [x] Documentation reflects the changes where applicable
* [x] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

### CODE REVIEW

**Review by sydney-runkle — APPROVED** [review]
Looks great! Approving now, will merge once we get that new core version ready :).

**Comment by NeevCohen** [comment]
@sydney-runkle Awesome! Thanks a lot! Just need to remember to remove the xfails from the tests 😄

**Comment by sydney-runkle** [comment]
@NeevCohen,

Once we merge this PR https://github.com/pydantic/pydantic/pull/9153, you can pull that change in and remove your xfails!

**Comment by sydney-runkle** [comment]
@NeevCohen, that PR is now merged!

File: tests/test_fields.py
Comment (sydney-runkle) [inline tests/test_fields.py:158]:
```suggestion
```

File: tests/test_fields.py
Comment (sydney-runkle) [inline tests/test_fields.py:149]:
```suggestion
```

**Comment by sydney-runkle** [comment]
@NeevCohen,

Hmm, looks like tests are still failing...

**Comment by sydney-runkle** [comment]
Going to go ahead and release 2.7 beta, but we can get this across the line this week + get it into the official release :).

**Comment by NeevCohen** [comment]
@sydney-runkle I figured out the issue. In core, I forgot to add the `corece_numbers_to_str` option in the `StringSchema` typed dict. Sorry for that, my bad.

I opened a PR fixing that issue, pydantic/pydantic-core#1262

**Comment by sydney-runkle** [comment]
Awesome, no worries at all. We can get that into the next `pydantic-core` release, which we'll include in 2.7!

**Comment by sydney-runkle** [comment]
@NeevCohen,

Could you pull in the changes from `main` once https://github.com/pydantic/pydantic/pull/9211 merges? Should be in just a few minutes :).

**Comment by sydney-runkle** [comment]
I'll review your other PR shortly - looking forward to it!! Just want to get 2.7 out first :).

**Pull request merged**
